Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/MemoryStatistics.h

```diff
@@ -89,6 +89,9 @@ namespace AZ
 
             /// The list of pools.
             AZStd::vector<Pool> m_pools;
+
+            /// Indicates if detailed memory statistics were captured
+            bool m_detailedCapture;
```

Comment:
Where is this used?

Reply:
I added this primarily as a countermeasure for cases where detailed reporting is requested but exactly 0 allocations are made. In this event, there is ambiguity over whether the pool stats are recorded without allocations, or if the allocations were included but simply didn't exist. While this isn't used today, I did not want a user of this class to assume that the data was available, so the API is self documenting in that sense.